api/clients/v2/verification/cert_verifier.go

```diff
@@ -107,6 +95,57 @@ func (cv *CertVerifier) CheckDACert(
 	return nil
 }
 
+// EstimateGasCheckDACert uses eth_estimateGas to estimate the gas requirements for a CheckDACert call.
+func (cv *CertVerifier) EstimateGasCheckDACert(
+	ctx context.Context,
+	cert coretypes.EigenDACert,
+) (uint64, error) {
+	// Normalize cert to V3
+	certV3 := NormalizeCertV3(cert)
+
+	certVerifierAddress, err := cv.addressProvider.GetCertVerifierAddress(
+		ctx,
+		certV3.ReferenceBlockNumber(),
+	)
+	if err != nil {
+		return 0, fmt.Errorf("get cert verifier address: %w", err)
+	}
+
+	certBytes, err := certV3.Serialize(coretypes.CertSerializationABI)
+	if err != nil {
+		return 0, fmt.Errorf("serialize cert: %w", err)
+	}
+
+	// Pack the checkDACert method call data
+	abi, err := certVerifierBinding.ContractEigenDACertVerifierMetaData.GetAbi()
+	if err != nil {
+		return 0, fmt.Errorf("get contract ABI: %w", err)
+	}
+
+	callData, err := abi.Pack("checkDACert", certBytes)
+	if err != nil {
+		return 0, fmt.Errorf("pack checkDACert call data: %w", err)
+	}
+
+	callMsg := ethereum.CallMsg{
+		To:   &certVerifierAddress,
+		Data: callData,
+	}
+
+	// Estimate gas using eth_estimateGas
+	gasEstimate, err := cv.ethClient.EstimateGas(ctx, callMsg)
+	if err != nil {
+		cv.logger.Error(
+			"eth_estimateGas",
+			zap.String("to", callMsg.To.Hex()),
+			zap.String("data", fmt.Sprintf("0x%x", callMsg.Data)),
```

Comment:
```suggestion
			"to", callMsg.To.Hex(),
			"data", fmt.Sprintf("0x%x", callMsg.Data),
```

Reply:
why are you using zap? Think its actually wrong here. We use slog most of the time, but also the interface for our logger.Error takes key value pairs, not a zap.String afaiu.

Reply:
Oh yikes that's actually super outdated. We moved to using slog as soon as it came out in go1.21

Reply:
updated in [254d7fb](https://github.com/Layr-Labs/eigenda/pull/1890/commits/254d7fb6027bf41789fc7f4f7a2d3ee4a531f9d9)